lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -0,0 +1,328 @@
+            reportProcessingResult.tacticalStoreLastEvaluatedKey(null);
+        }
+    }
+
+    private void processUsersToFindLOCAndUpdateDb(
+            ReportProcessingRequest reportProcessingRequest,
+            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult,
+            Context context)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException,
+                    BatchDeleteException, StopBeforeLambdaTimeoutException {
+        if (reportProcessingRequest.continueUserIdentityScan()) {
+            LOGGER.info(
+                    LogHelper.buildLogMessage(
+                            "Processing unique users to update their aggregate identities."));
+
+            List<ReportUserIdentityItem> totalReportUserIdentityItems = new ArrayList<>();
+            for (var page :
+                    reportUserIdentityScanDynamoDataStore.getItems(
+                            reportProcessingRequest.userIdentitylastEvaluatedKey(),
+                            ATTR_NAME_USER_ID,
+                            "identity",
+                            "migrated")) {
+                var userIds = page.items().stream().map(ReportUserIdentityItem::getUserId).toList();
+                LOGGER.info(
+                        LogHelper.buildLogMessage(
+                                String.format(
+                                        "Checking user identity for total (%s) users.",
+                                        userIds.size())));
+                List<ReportUserIdentityItem> reportUserIdentityItems = new ArrayList<>();
+                for (String userId : userIds) {
+                    evaluateUserIdentityDetail(reportUserIdentityItems, userId);
+                }
+                LOGGER.info(LogHelper.buildLogMessage("Updating processed user's identity."));
+                reportUserIdentityScanDynamoDataStore.createOrUpdate(reportUserIdentityItems);
+
+                aggregateAndPersistIdentities(reportUserIdentityItems);
+
+                totalReportUserIdentityItems.addAll(reportUserIdentityItems);
+                reportProcessingResult.userIdentitylastEvaluatedKey(page.lastEvaluatedKey());
+
+                try {
+                    checkLambdaRemainingExecutionTime(context);
+                } catch (StopBeforeLambdaTimeoutException e) {
+                    reportProcessingResult.users(totalReportUserIdentityItems);
+                    throw e;
+                }
+            }
+            LOGGER.info(LogHelper.buildLogMessage("User identity check scan completed."));
+            reportProcessingResult.userIdentitylastEvaluatedKey(null);
+            reportProcessingResult.users(totalReportUserIdentityItems);
+        }
+    }
+
+    private void evaluateUserIdentityDetail(
+            List<ReportUserIdentityItem> reportUserIdentityItems, String userId)
+            throws CredentialParseException, HttpResponseExceptionWithErrorBody, ParseException {
```

Comment:
(Feels like it'd be more natural for this method to return the evaluated identity, rather than append to a list?)